Adding some doc examples - Issues258

allmightyspiff opened the pull request:
some basic examples on the vs and sever managers mostly.

sudorandom commented on SoftLayer/managers/hardware.py:
```diff
@@ -288,6 +335,65 @@ def place_order(self, **kwargs):
                                   interfaces
         :param list extras: List of extra feature names
+
+        .. warning::
```
This example/description is no longer true. place_order uses the simplified hardware ordering and is able to take more-or-less human readable values. See the new `slcli server create-options` and `slcli server create` commands to see how it works now.

sudorandom commented on SoftLayer/managers/hardware.py:
```diff
@@ -80,6 +94,10 @@ def cancel_metal(self, hardware_id, immediate=False):
                                scheduled to cancel on the anniversary date.
+        Example::
+
+            result = mgr.cancel_metal(hardware_id=1234)
```
cancel_metal() is being removed in https://github.com/softlayer/softlayer-python/pull/524 because it doesn't seem needed any longer as cancel_hardware() can serve the same purpose now.

sudorandom commented on SoftLayer/managers/hardware.py:
```diff
@@ -46,6 +55,11 @@ def cancel_hardware(self, hardware_id, reason='unneeded', comment='',
+
+            # Cancels hardware id 1234
+            result = mrg.cancel_hardware(hardware_id=1234)
+            print result
```
Try to avoid python-version-specific examples

allmightyspiff commented:
Removed the examples for place order, when I figure out the new hotness I'll add a more relevant  example. 
Removed the print statements, they were superfluous anyway

I removed the cancel_metal example

underscorephil commented on SoftLayer/managers/hardware.py:
```diff
@@ -26,6 +26,15 @@ class HardwareManager(utils.IdentifierMixin, object):
                                               If none is provided, one will be
                                               auto initialized.
+    Example::
+       # Initialize the Manager.
```
A space is required after Example::

underscorephil commented on SoftLayer/managers/hardware.py:
```diff
@@ -259,6 +287,15 @@ def change_port_speed(self, hardware_id, public, speed):
+            A port speed of 0 will disable the interface.
+
+        Example::
+            #change the Public interface to 10Mbps on instance 12345
```
A space is required after Example::

underscorephil commented on SoftLayer/managers/hardware.py:
```diff
@@ -503,6 +540,10 @@ def edit(self, hardware_id, userdata=None, hostname=None, domain=None,
         :param string notes: notes about this particular hardware
 
+        Example::
+            # Change the hostname on instance 12345 to 'something'
```
A space is required after Example::

underscorephil commented on SoftLayer/managers/ssl.py:
```diff
@@ -11,6 +11,17 @@ class SSLManager(object):
     :param SoftLayer.API.Client client: an API client instance
+
+    Example::
+       # Initialize the Manager.
```
A blank line is required after Example::

underscorephil commented on SoftLayer/managers/vs.py:
```diff
@@ -23,6 +23,17 @@ class VSManager(utils.IdentifierMixin, object):
                                               auto initialized.
+
+    Example::
+           # Initialize the VSManager.
```
A blank line is required after Example::

underscorephil commented on SoftLayer/managers/vs.py:
```diff
@@ -384,6 +388,10 @@ def wait_for_ready(self, instance_id, limit, delay=1, pending=False):
                              provisioning or reloads such as monitoring.
+
+        Example::
+            # Will return once vsi 12345 is ready, or after 10 checks
```
A blank line is required after Example::

underscorephil commented on SoftLayer/managers/vs.py:
```diff
@@ -423,6 +431,27 @@ def verify_create_instance(self, **kwargs):
         See :func:`create_instance` for a list of available options.
+
+        Example::
+            new_vsi = {
```
A blank line is required after Example::

underscorephil commented on SoftLayer/managers/vs.py:
```diff
@@ -459,6 +488,30 @@ def create_instance(self, **kwargs):
+            This will add charges to your account
+
+        Example::
+            new_vsi = {
```
A blank line is required after Example::

underscorephil commented on SoftLayer/managers/vs.py:
```diff
@@ -471,6 +524,39 @@ def create_instances(self, config_list):
+        .. warning::
+            This will add charges to your account
+
+        Example::
```
A blank line is required after Example::

underscorephil commented on SoftLayer/managers/vs.py:
```diff
@@ -491,6 +577,15 @@ def change_port_speed(self, instance_id, public, speed):
+            A port speed of 0 will disable the interface.
+
+        Example::
+            #change the Public interface to 10Mbps on instance 12345
```
A blank line is required after Example::

underscorephil replied:
A blank line is required after Example::

underscorephil commented on SoftLayer/managers/vs.py:
```diff
@@ -535,7 +630,12 @@ def edit(self, instance_id, userdata=None, hostname=None, domain=None,
+        :returns: bool -- True or an Exception
 
+        Example::
+            # Change the hostname on instance 12345 to 'something'
```
A blank line is required after Example::

underscorephil replied:
A blank line is required after Example::

underscorephil commented on SoftLayer/managers/vs.py:
```diff
@@ -563,6 +663,11 @@ def rescue(self, instance_id):
         :param integer instance_id: the instance ID to rescue
+        :returns: bool -- True or an Exception
+
+        Example::
```
A blank line is required after Example::

underscorephil commented on SoftLayer/managers/hardware.py:
```diff
@@ -46,6 +55,10 @@ def cancel_hardware(self, hardware_id, reason='unneeded', comment='',
+        Example::
+
+            # Cancels hardware id 1234
+            result = mrg.cancel_hardware(hardware_id=1234)
```
Should this be mgr?

underscorephil commented on SoftLayer/managers/hardware.py:
```diff
@@ -181,6 +200,11 @@ def get_hardware(self, hardware_id, **kwargs):
+
+            object_mask = "mask[id,networkVlans[vlanNumber]]"
+            # Object masks are optional
+            result = mrg.get_hardware(hardware_id=1234,mask=object_mask)
```
Should this be mgr?
